interview-questions.md

```diff
@@ -7,53 +7,55 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: I'm not sure what a relational database is, I'll have to research that question afterwards. I do know that databases in general are there to store large amounts of information. For instance you can make a database that stores all the user information of people who sign up to your website or other user based product.
 
-  Researched answer:
+  Researched answer: "A relational database is a collection of information that organizes data points with defined relationships for easy access. In the relational database model, the data structures -- including data tables, indexes and views -- remain separate from the physical storage structures, enabling database administrators to edit the physical data storage without affecting the logical data structure." There are multiple databases that can be used, but the top three from my research, besides relational databases are hierarchical database systems, network database systems, and object-oriented database systems.
+
+  https://www.techtarget.com/searchdatamanagement/definition/relational-database
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: Primary keys refer to spefic id's inside of a database that can't have the same id number. Starting at 1 and going until no more user are assigned a number, these numbers help when it comes to associating specfic information to a specific id. They're also important because it allows the developer to update specific information based off that one primary key, instead of updating all user information across the board. 
 
-  Researched answer:
+  Researched answer: Primary keys are located inside of a column with the database table. Along with that, the database table can't have columns without a primary key inside of it, nor can a table contain a null value with a primary key.
 
 
 
 ## Looking Ahead: Terms for Next Week
 Research and define the following terms to the best of your ability.
 
-1. RESTful routes:
+1. RESTful routes: RESTful routes provide a path from the HTTP verbs to the controller CRUD action. THis is helpful because it doesn't require a URL to redirect the webpage to the correct site.
 
-2. JSON:
+2. JSON: JSON stands for JavaScript Object Notation. JSON is used for storing and transmitting data between a server and a web page.
 
-3. ERB:
+3. ERB: Stands for, Embedded Ruby. ERB allows ruby users to generate any kind of text from templates.
 
-4. Params:
+4. Params: Params in rails refer to when parameters are being passed to the controller by the 'get' or 'post' request.
 
-5. API:
+5. API: Stands for, Application Program Interface. API's help transfer data from an outside sources through the internet, into the local application
```

Comment:
You CRUSHED this section Jonathan!!  This is exactly what I am looking for in an interview.  Excellent work!